File: 2026-1/sesion-de-clase-24/java/src/main/java/ejercicio2/HiloConPausa.java
package ejercicio2;

public class HiloConPausa extends Thread {
    private static final long PAUSA_MILLIS = 5000L;

    @Override
    public void run() {
        try {
            Thread.sleep(PAUSA_MILLIS);
        }
        catch (InterruptedException ex) {
            Thread.currentThread().interrupt();
            return;
        }

        System.out.println("Hilo '" + getName() + "'");
    }
}

File: 2026-1/sesion-de-clase-24/net/sesion-de-clase-24/Ejercicio2/HiloConPausa.cs
using System;
using System.Threading;

namespace Ejercicio2;

public class HiloConPausa
{
    private const int PausaMillis = 5000;

    public void Run()
    {
        try
        {
            Thread.Sleep(PausaMillis);
        }
        catch (ThreadInterruptedException)
        {
            Thread.CurrentThread.Interrupt();
            return;
        }

        Console.WriteLine($"Hilo '{Thread.CurrentThread.Name}'");
    }
}
